source/slang/slang-emit-spirv.cpp

```diff
@@ -4333,10 +4366,10 @@ struct SPIRVEmitContext
 
     void maybeEmitPointerDecoration(SpvInst* varInst, IRInst* inst)
     {
-        auto ptrType = as<IRPtrType>(inst->getDataType());
+        auto ptrType = as<IRPtrType>(unwrapArray(inst->getDataType()));
         if (!ptrType)
             return;
-        if (ptrType->getAddressSpace() == SpvStorageClassPhysicalStorageBuffer)
+        if (addressSpaceToStorageClass(ptrType->getAddressSpace()) == SpvStorageClassPhysicalStorageBuffer)
```

Comment:
can this change to 
`ptrType->getAddressSpace() == UserPointer` ?

Reply:
At this stage, a ptr type can either have AddressSpace::UserPointer or SpvStorageClassPhysicalStorageBuffer, we want both cases to map to PhysicalStorageBuffer, so this is how we can ensure that.